### PR TITLE
Case sensitive MS SQL database operations fail because of uppercase system table references

### PR DESCRIPTION
**Pull request by SLOBYYYY**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)? (**see below**)
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions? (**not new but changed the existing ones**)
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)? (**this is my first PR but hopefully yes**)

`npm run test-mssql` fails but it failed for me before my changes. After my changes it only fails at the same place:

```
> sequelize@5.0.0-beta.3 test-integration /tmp/sequelize
> mocha $npm_package_options_mocha "test/integration/**/*.test.js"

FIXME: I want to be supported in this dialect as well :-(


  [MSSQL] Alias
      1) "before each" hook for "should uppercase the first letter in alias getter, but not in eager loading"
      2) "after each" hook for "should uppercase the first letter in alias getter, but not in eager loading"


            0 passing (50ms)
            2 failing
```

### Description of change

Please read PR [#9336](https://github.com/sequelize/sequelize/issues/9336) for description.

### CODE REVIEW

File: lib/dialects/mssql/data-types.js [hunk 1]
```diff
@@ -135,9 +135,13 @@ module.exports = BaseTypes => {
   inherits(DATE, BaseTypes.DATE);
 
   DATE.prototype.toSql = function toSql() {
-    return 'DATETIMEOFFSET';
```
Comment (sushantdhiman) [inline lib/dialects/mssql/data-types.js]: ?

File: lib/dialects/mssql/data-types.js [hunk 2]
```diff
@@ -135,9 +135,13 @@ module.exports = BaseTypes => {
...
+    return 'DATETIME';
   };
 
+  DATE.prototype._stringify = function _stringify(date) {
```
Comment (sushantdhiman) [inline lib/dialects/mssql/data-types.js]: ???

File: lib/dialects/mssql/data-types.js (hunk 1 above)
Comment (SLOBYYYY) [reply lib/dialects/mssql/data-types.js]: Sorry for the mixup and confusion: I started doing changes for a project of ours in the first place. The PR contained the initial changes we needed but we went further. The mistake was that I forgot that if I commit and push the new changes, this PR will also be updated.

Moved the new -and unrelated to this PR- changes to a different branch and reverted the changes you questioned.

File: lib/dialects/mssql/data-types.js (hunk 2 above)
Comment (SLOBYYYY) [reply lib/dialects/mssql/data-types.js]: Same applies to this comment as I previously mentioned.

**Pull request merged**